Sources/Misc/OperationDispatcher.swift

```diff
@@ -16,27 +16,52 @@ import Foundation
 
 class OperationDispatcher {
 
-    private let mainQueue = DispatchQueue.main
-    private let workerQueue = DispatchQueue(label: "OperationDispatcherWorkerQueue")
+    private let mainQueue: DispatchQueue = .main
+    private let workerQueue: DispatchQueue = .init(label: "OperationDispatcherWorkerQueue")
     private let maxJitterInSeconds: Double = 5
 
     static let `default`: OperationDispatcher = .init()
 
-    func dispatchOnMainThread(_ block: @escaping () -> Void) {
+    func dispatchOnMainThread(_ block: @escaping @Sendable () -> Void) {
```

Comment:
Should this be made private now? Is there any scenario where we should choose to go with main thread and not main actor even when Swift Concurrency is available?

Reply:
Eventually we can get rid of this, but that seems like a bigger change. For existing code that does not use the new Swift "structured" concurrency, we might not want to change the semantics if we don't need to.

There's still a few places using `dispatchOnMainThread` that we might not want to change until we use `@MainActor` (or other custom `actor`s) in more places.

Reply:
Let men now what you think, we can migrate more of these to `dispatchOnMainActor` in a following PR.

Reply:
👍